src/service/prices/index.ts

```
@@ -0,0 +1,606 @@
+
+  /**
+   * Maximum number of tokens to process in a single batch during price updates.
+   * Balances update efficiency with Stellar network and Redis load.
+   */
+  private static readonly TOKEN_UPDATE_BATCH_SIZE = 150;
+
+  /**
+   * Maximum number of tokens to fetch and track prices for initially.
+   * Limits the total number of tokens to manage system resource usage.
+   */
+  private static readonly INITIAL_TOKEN_COUNT = 1000;
+
+  /**
+   * Stellar Expert API endpoint for fetching all tradable assets.
+   */
+  private static readonly STELLAR_EXPERT_ALL_ASSETS_URL =
+    "https://api.stellar.expert/explorer/public/asset";
+
+  /**
+   * Base URL for Stellar Expert API calls.
+   * Used to construct pagination URLs when fetching multiple pages of assets.
+   */
+  private static readonly STELLAR_EXPERT_BASE_URL =
+    "https://api.stellar.expert";
+
+  private readonly logger: Logger;
+  private readonly server: StellarSdk.Horizon.Server;
+
+  /**
+   * Creates a new PriceClient instance.
+   *
+   * @param logger - The logger instance for logging events and errors
+   * @param redisClient - Optional Redis client with time series support for caching prices
+   */
+  constructor(
+    logger: Logger,
+    private readonly redisClient?: RedisClientWithTS,
+  ) {
+    this.logger = logger;
+    const Sdk = getSdk(StellarSdkNext.Networks.PUBLIC);
+    const { Horizon } = Sdk;
+    this.server = new Horizon.Server(NETWORK_URLS.PUBLIC, {
+      allowHttp: true,
+    });
+  }
+
+  /**
+   * Retrieves the current price and 24-hour price change percentage for a token.
+   * If the token is not in the cache, it also adds it to the cache.
+   *
+   * @param token - The token identifier in format "code:issuer" or "native" for native asset
+   * @returns The token price data or null if price cannot be retrieved
+   */
+  getPrice = async (token: string): Promise<TokenPriceData | null> => {
+    if (!this.redisClient) {
+      return null;
```

Should we return an error here so that consuming clients don't mistake this for a missing token price?